### PR TITLE
redact spec api

### PR DESCRIPTION
**Pull request by laverya**
implements https://github.com/replicatedhq/kots/pull/569

### CODE REVIEW

File: kotsadm/pkg/handlers/redact.go [hunk 1]
```diff
@@ -159,3 +187,271 @@ func GetRedact(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		logger.Error(err)
+		getMetadataResponse.Error = "failed to parse authorization header"
+		JSON(w, 401, getMetadataResponse)
+		return
+	}
+
+	// we don't currently have roles, all valid tokens are valid sessions
+	if sess == nil || sess.ID == "" {
+		getMetadataResponse.Error = "failed to parse authorization header"
```
Comment (divolgin) [inline kotsadm/pkg/handlers/redact.go:214]: maybe a different message here, because this one is already used

Comment (laverya) [reply kotsadm/pkg/handlers/redact.go:214]: sounds like a good idea

File: kotsadm/pkg/redact/redact.go [hunk 2]
```diff
@@ -132,10 +173,278 @@ func SetRedactSpec(spec string) (string, error) {
...
+	}
+
+	newConfigMap, err := clientset.CoreV1().ConfigMaps(os.Getenv("POD_NAMESPACE")).Update(configMap)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to update configmap")
+	}
+	return newConfigMap, nil
+}
+
+func getSlug(name string) string {
```
Comment (divolgin) [inline kotsadm/pkg/redact/redact.go:357]: I don't think this function is 1-to-1.  why is this needed?

Comment (laverya) [reply kotsadm/pkg/redact/redact.go:357]: I thought there were restrictions on the keys a configmap could have in k8s
if there aren't, I'd love to drop slugs entirely
